pkg/wekafs/apiclient/apiendpoint.go

```diff
@@ -146,6 +156,10 @@ func (eps *ApiEndPoints) UpdateEndpointsFromAddresses(ctx context.Context, addre
 			eps.RemoveEndpoint(key)
 		}
 	}
+	if len(eps.endpoints) == 0 {
+		return status.Error(codes.Internal, "no endpoints provided")
```

Comment:
the error is not informative enough. What endpoints? Also, it could be that endpoints were provided but not valid.